docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,298 @@
+- The EdgeX MessageBus using Redis is a Secure MessageBus, thus the addition of the `AuthMode` and `SecretName` settings which allow the credentials to be pulled from the service's SecretStore. See the [Secure MessageBus](../../../security/Ch-Secure-MessageBus) secure for more details.
+
+##### External MQTT
+
+If your Application service is using the **External MQTT** trigger do the following:
+
+1. Move your existing `MqttBroker` configuration under the `Trigger` section (renaming it to `ExternalMqtt`)
+2. Move your `SubscribeTopic` (renaming it to `SubscribeTopics`) under the `ExternalMqtt` section.
+3. Move your `PublishTopic` under the `ExternalMqtt` section.
+
+!!! example "Example - External MQTT trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="external-mqtt"
+      [Trigger.ExternalMqtt]
+      Url = "tcp://broker.hivemq.com:1883"
+      SubscribeTopics = "edgex-trigger"
+      PublishTopic = "edgex-trigger-response"
+      ClientId = "app-my-service"
+      ConnectTimeout = "30s"
+      AutoReconnect = false
+      KeepAlive = 60
+      QoS = 0
+      Retain = false
+      SkipCertVerify = false
+      SecretPath = ""
+      AuthMode = "none"
+    ```
+
+##### HTTP
+
+The HTTP trigger configuration has not changed beyond the renaming of `Binding` to `Trigger`.
+
+!!! example "Example - HTTP trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="http"
+    ```
+
+### Code
+
+The first changes you will encounter in your code are that the `AppFunctionsSDK` and `Context` structs have be abstracted into the new `ApplicationService` and `AppFunctionContext` APIs. See the [Application Service API](../ApplicationServiceAPI) and [App Function Context API](../AppFunctionContextAPI) sections for complete details on these new APIs.
+
+#### main()
+
+The following changes to your `main()` function will be necessary.
+
+##### Create and Initialize
+
+Your `main()` will change to use a factory function to create and initialize the Application Service instance, rather than create instance of `AppFunctionsSDK` and call Initialize()` 
+
+!!! example "Example - Create Application Service instance"
+
+    ```go
+        const serviceKey = "app-myservice"
+        ...
+    
+        service, ok := pkg.NewAppService(serviceKey)
+        if !ok {
+            os.Exit(-1)
+        }
+    ```
+
+!!! example "Example - Create Application Service instance with Target Type specified"
+
+    ```go
+        const serviceKey = "app-myservice"
+        ...
+    
+        service, ok := pkg.NewAppServiceWithTargetType(serviceKey, &[]byte{})
+        if !ok {
+            os.Exit(-1)
+        }
+    ```
+
+Since the factory function logs all errors all you need to do is exit if it returns `false`. 
```

Comment:
You might want to add a comma in this since sentence.

Reply:
added